### PR TITLE
[DOCS] mention automatic cancellation in search docs

### PR DESCRIPTION
**Pull request by javanna**
Add a small note around automatic search cancellation which was missed as part of #43332

Relates to #43332

### CODE REVIEW

File: docs/reference/search.asciidoc [hunk 1]
```diff
@@ -125,9 +125,12 @@ Setting this value to `-1` resets the global search timeout to no timeout.
...
 === Search Cancellation
 
-Searches can be cancelled using standard <<task-cancellation,task cancellation>>
-mechanism. By default, a running search only checks if it is cancelled or
-not on segment boundaries, therefore the cancellation can be delayed by large
+Searches are automatically cancelled when the http connection that they came
+in from gets closed. It is fundamental that the http client sending requests
+closes connections whenever requests time out or are aborted. Automatic
+cancellation relies on standard  <<task-cancellation,task cancellation>>
+mechanism. By default, a running search only checks if it is cancelled or not
```
Comment (jimczi) [inline docs/reference/search.asciidoc]: This is unrelated to this change but the default value for low_level_cancellation is now true so the doc is not accurate. We should remove this sentence entirely: `By default, a running search only checks if it is cancelled or not...`.

File: docs/reference/search.asciidoc [hunk 2]
```diff
@@ -125,9 +125,12 @@ Setting this value to `-1` resets the global search timeout to no timeout.
 [[global-search-cancellation]]
 === Search Cancellation
 
-Searches can be cancelled using standard <<task-cancellation,task cancellation>>
-mechanism. By default, a running search only checks if it is cancelled or
-not on segment boundaries, therefore the cancellation can be delayed by large
+Searches are automatically cancelled when the http connection that they came
```
Comment (jimczi) [inline docs/reference/search.asciidoc]: I find the sentence hard to read, maybe something like: `when the http connection used to perform the request is closed by the client` ?

**Review by jimczi — COMMENTED** [review]
I left some minor comments

**Comment by javanna** [comment]
@jimczi can you have another look?

**Review by jimczi — APPROVED** [review]
LGTM

**Review by giladgal — APPROVED** [review]

**Pull request merged**